libs/utils/src/filter.c

```diff
@@ -725,4 +724,71 @@ const char* celix_filter_findAttribute(const celix_filter_t *filter, const char
         }
     }
     return result;
+}
+
+static bool hasMandatoryEqualsValueAttribute(const celix_filter_t *filter, const char *attribute, bool negated, bool optional) {
+    bool equalsValueAttribute = false;
+
+    if (filter != NULL && attribute != NULL) {
+        if (filter->operand == CELIX_FILTER_OPERAND_AND || filter->operand == CELIX_FILTER_OPERAND_OR || filter->operand == CELIX_FILTER_OPERAND_NOT) {
+            size_t size = celix_arrayList_size(filter->children);
```

Comment:
celix_arrayList_size returns a `int` not `size_t`. Note that maybe a `size_t` should have been better, but to prevent breaking the API is kept at `int`.

So imo should should be:
```
int size = celix_arrayList_size(filter->children);
for (int i = 0; i < size; ++i) {
 ...
```

Reply:
changed to int.